### ISSUE TITLE
[Data Explorer] Make Discover chart and table interactive with top nav and Querybar

### ISSUE THREAD

**Issue by ananzh** [opensearch-project/OpenSearch-Dashboards#4632]
Blocked by https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4226

**Comment by ananzh** [comment]
This feature is done and merged into feature branch.
